packages/edit-post/src/components/header/mode-switcher/index.js

```diff
@@ -45,7 +45,7 @@ function ModeSwitcher() {
 			isCodeEditingEnabled:
 				select( editorStore ).getEditorSettings().codeEditingEnabled,
 			isEditingTemplate:
-				select( editorStore ).getRenderingMode() === 'template-only',
+				select( editorStore ).getCurrentPostType() === 'wp_template',
```

Comment:
This one hides the code editor when editing the template in the post editor (see #37359).

I think it might be ok to reintroduce the code editor as it seems to work fine now in my testing (cc @Mamaduka what do you think?)

It could be one to roll out into a small separate PR.

Reply:
Yeh, let's look at that in a follow-up PR.

Reply:
Agreed with Dan, it seems very random.

Reply:
I started on this in a separate PR -  [Unify the Editor Mode Switcher, and enable it for template editing in the post editor](https://github.com/WordPress/gutenberg/pull/57868#top)

However, it seems like the changes in this PR fix issues with the code editor mode for template mode in the post editor, so I'll keep that PR on hold until this one is shipped. As a short term step I'll remove the `isEditingTemplate` usage in this file.